apps/meteor/tests/e2e/admin-room.spec.ts

```diff
@@ -0,0 +1,116 @@
+import { faker } from '@faker-js/faker';
+import { Page } from '@playwright/test';
+
+import { Users } from './fixtures/userStates';
+import { createTargetChannel, createTargetPrivateChannel } from './utils';
+import { expect, test } from './utils/test';
+
+test.use({ storageState: Users.admin.state });
+
+test.describe.serial('admin-rooms', () => {
+	let adminPage: Page;
+	let channel: string;
+	let privateRoom: string;
+
+	test.beforeEach(async ({ page }) => {
+		await page.goto('/admin/rooms');
+	});
+
+	test.beforeAll(async ({ browser, api }) => {
+		adminPage = await browser.newPage({ storageState: Users.admin.state });
+		await adminPage.goto('/home');
+		await adminPage.waitForSelector('[data-qa-id="home-header"]');
+
+		channel = await createTargetChannel(api);
+		privateRoom = await createTargetPrivateChannel(api);
+	});
+	test('should display the Rooms Table', async ({ page }) => {
+		await expect(page.locator('[data-qa-type="PageHeader-title"]')).toContainText('Rooms');
+	});
+
+	test('should filter room by name', async ({ page }) => {
+		const input = page.locator('[data-qa-id="AdminRoomSearchInput"]');
+
+		await input.click();
+		await input.fill(channel);
+
+		await expect(page.locator(`[qa-room-name="${channel}"]`)).toBeVisible();
+	});
+
+	test('should filter rooms by type', async ({ page }) => {
+		const dropdown = page.locator('[data-qa-id="AdminRoomDropdownInput"]');
+		await dropdown.click();
+
+		const privateOption = page.locator('text=Private channels');
+
+		await privateOption.waitFor();
+		await privateOption.click();
+
+		await expect(page.locator('text=Private Channel').first()).toBeVisible();
+	});
+
+	test('should filter rooms by type and name', async ({ page }) => {
+		const input = page.locator('[data-qa-id="AdminRoomSearchInput"]');
+
+		await input.click();
+		await input.fill(privateRoom);
+
+		const dropdown = page.locator('[data-qa-id="AdminRoomDropdownInput"]');
+		await dropdown.click();
+
+		const privateOption = page.locator('text=Private channels');
+
+		await privateOption.waitFor();
+		await privateOption.click();
+
+		await expect(page.locator(`[qa-room-name="${privateRoom}"]`)).toBeVisible();
+	});
+
+	test('should be empty in case of the search does not find any room', async ({ page }) => {
+		const input = page.locator('[data-qa-id="AdminRoomSearchInput"]');
+
+		await input.click();
+		const wrongChannel = faker.string.alpha(10);
```

Comment:
We could change the name to "nonExistingChannel" since the channel is not "wrong". This might help clarifies that the channel does not exist rather than being inherently "wrong."